docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -0,0 +1,1362 @@
+
+```golang
+type MPISpec struct {
+    // Number of Procs per Node.
+    NumProcPerNode int `json:"numProcPerNode,omitempty"`
+
+    // MPI Implementation to create appropriate host-files.
+    // Can be one of OpenMPI, Intel, or MPICH.
+    MPIImplementation MPIImplementation `json:"mpiImplementation,omitempty"`
+
+    // Directory where SSH keys are mounted.
+    SSHAuthMountPath string `json:"SSHAuthMountPath,omitempty"`
+}
+
+type MPIImplementation string
+
+const (
+    MPIImplementationOpenMPI MPIImplementation = "OpenMPI"
+    MPIImplementationIntel   MPIImplementation = "Intel"
+    MPIImplementationMPICH   MPIImplementation = "MPICH"
+)
+```
+
+### Supported Runtimes by Community
+
+Kubeflow community are planning to support the following runtimes.
+
+#### PyTorch Distributed Runtime
+
+Initially, we will maintain only multi-node multi-worker runtime and PyTorch Elastic.
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: torch-distributed-multi-node
+spec:
+  mlSpec:
+    torch:
+      numProcPerNode: 5
+  replicatedJobs:
+    - name: node
+      template:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: trainer
+                  image: docker.io/kubeflow/pytorch-mnist
+                  env:
+                    - name: MASTER_ADDR
+                      value: "pytorch-node-0-0.pytorch"
+                    - name: MASTER_PORT
+                      value: 29400
+                  command:
+                    - torchrun train.py
+```
+
+Example of usage:
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: TrainJob
+metadata:
+  name: torch-distributed-multi-node
+  namespace: tenant-alpha
+spec:
+  trainingRuntimeRef:
+    name: pytorch-distributed
+  trainerConfig:
+    resourcesPerNode:
+      requests:
+        nvidia.com/gpu: 1
+    args:
+      - num-epochs=5
+```
+
+#### PyTorch Elastic Runtime
+
+Training runtime for PyTorch Elastic:
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: torch-distributed-elastic
+spec:
+  mlSpec:
+    torchSpec:
+      elasticPolicy:
+        minNodes: 5
+        maxNodes: 10
+        metrics:
+          - type: Resource
+            resource:
+              name: cpu
+              target:
+                type: Utilization
+                averageUtilization: 80
+  replicatedJobs:
+    - name: node
+      template:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: trainer
+                  image: docker.io/kubeflow/pytorch-mnist
+                  env:
+                    - name: MASTER_ADDR
+                      value: "pytorch-node-0-0.pytorch"
+                    - name: MASTER_PORT
+                      value: 29400
+                  command:
+                    - torchrun train.py
+```
+
+#### Additional PyTorch Runtimes
+
+The following runtimes can be maintained in the future.
```

Comment:
Should this be part of this initial KEP if it's TBA / for the future/ maybe very likely to change?